modules/nf-core/jvarkit/vcffilterjdk/main.nf

```diff
@@ -0,0 +1,75 @@
+process JVARKIT_VCFFILTERJDK {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/jvarkit:2024.08.25--hdfd78af_1':
+        'biocontainers/jvarkit:2024.08.25--hdfd78af_1' }"
+
+    input:
+    tuple val(meta),  path(vcf), path(tbi), path(regions_file)
+    tuple val(meta2), path(fasta)
+    tuple val(meta3), path(fai)
+    tuple val(meta4), path(dict)
+    tuple val(meta5), path(code)
+    output:
+    tuple val(meta), path("*.${extension}"), emit: vcf
+    tuple val(meta), path("*.tbi")         , emit: tbi, optional: true
+    tuple val(meta), path("*.csi")         , emit: csi, optional: true
+    path "versions.yml"                    , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args1        = task.ext.args1 ?: ''
+    def args2        = task.ext.args2 ?: ''
+    def args3        = task.ext.args3 ?: ''
+    def prefix       = task.ext.prefix ?: "${meta.id}"
+    def script_file  = code?"--script \"${code}\"":""
+    def regions_file = regions_file? (tbi ? " --regions-file" : " --targets-file")+" \"${regions_file}\" ":""
+
+    extension =     getVcfExtension(args3); /* custom function, see below */
```

Comment:
I would not use a function here, just put it straight in

Reply:
@jfy133 this statement contradicts what I've been told here https://nfcore.slack.com/archives/CJRH30T6V/p1725629116371819

Reply:
_grumble grumble_ @nvnieuwk have you being saying yes to others on this? I thought we were trying to avoid such complexity in modules as an approximate rule ?

Reply:
I thought it seemed reasonable :grin: I didn't know about us trying to avoid this, my bad. If that's the case than we shouldn't do this

Reply:
That's what's happening here right? (sorry didn't take a closer look to it yet, it's a chaotic day) :grin: